tests/sync/genesis/test_genesis_voting.py

```diff
@@ -0,0 +1,84 @@
+import time
+import pytest
+from pytest_testconfig import config as testconfig
+
+import tests.queries as q
+from tests.sync.test_sync import new_client_in_namespace
+from tests.test_bs import get_conf, setup_bootstrap, start_poet, add_curl
+from tests.utils import print_hits_entry_count
+
+# ==============================================================================
+#    TESTS
+# ==============================================================================
+
+
+# set a X nodes cluster
+# sleep in order to enable a block to be delivered
+# validate a block has been delivered
+# add an additional client while still in genesis
+# sleep until layer 4
+# validate no errors occurred
+# validate new node didn't receive any new blocks before being synced
+# @pytest.mark.xfail(strict=True, reason="new pod creates blocks before being synced")
+def test_unsync_while_genesis(init_session, setup_bootstrap, start_poet, add_curl):
+    time_before_first_block = 70
+    layers_to_wait = 4
+
+    layer_duration = int(testconfig['client']['args']['layer-duration-sec'])
+    bs_info = setup_bootstrap.pods[0]
+    cspec = get_conf(bs_info, testconfig['client'], None, setup_bootstrap.pods[0]['pod_ip'])
+
+    # Create a cluster of nodes
+    _ = new_client_in_namespace(testconfig['namespace'], setup_bootstrap, cspec, 9)
+
+    # Sleep to enable block creation
+    print(f"sleeping for {time_before_first_block} seconds in order to enable blocks to be published\n")
+    time.sleep(time_before_first_block)
+
+    # Validate a block was published
+    nodes_published_block, _ = q.get_blocks_per_node_and_layer(init_session)
+    if not nodes_published_block:
+        assert 0, f"no blocks were published during the first {time_before_first_block} seconds"
+
+    # Create a new node in cluster
+    unsynced_cl = new_client_in_namespace(testconfig['namespace'], setup_bootstrap, cspec, 1)
+
+    # Sleep until layers_to_wait layer, default is 4
+    print(f"sleeping for {layer_duration * layers_to_wait} seconds\n")
+    time.sleep(layer_duration * layers_to_wait)
+
+    # Found by Almogs: "validate votes failed" error has occurred following a known bug
+    print("validating no 'validate votes failed' messages has arrived")
+    hits_val_failed = q.get_all_msg_containing(init_session, init_session, "validate votes failed")
+    assert hits_val_failed == [], 'got a "validate votes" failed message'
+    print("validation succeeded")
+
+    # Get the msg when app started on the late node
+    app_started_hits = q.get_app_started_msgs(init_session, unsynced_cl.pods[0]["name"])
+    if not app_started_hits:
+        assert 0, f"app did not start for new node after {layers_to_wait} layers"
```

Comment:
although not important and syntax related, does `assert 0` mean fail? and if so, can it be replaced with `assert False` ?

Reply:
Yes, it fails the test and prints the message attached.
I don't mind changing it, thought it would be more easy to read than in one line

Reply:
fixed in 5c6b63aabad9a2849d5c4dcc7755c9ab121a0579